src/Components/Web.JS/src/Platform/Circuits/CircuitStreamingInterop.ts

```diff
@@ -0,0 +1,53 @@
+import { HubConnection } from '@microsoft/signalr';
+
+export function sendJSDataStream(connection: HubConnection, data: ArrayBufferView, streamId: string, chunkSize: number) {
+    // Run the rest in the background, without delaying the completion of the call to sendJSDataStream
+    // otherwise we'll deadlock (.NET can't begin reading until this completes, but it won't complete
+    // because nobody's reading the pipe)
+    setTimeout(async () => {
+        const maxMillisecondsBetweenAcks = 500;
+        let numChunksUntilNextAck = 5;
+        let lastAckTime = new Date().valueOf();
+        try {
+            let position = 0;
+            let chunkId = 0;
+
+            // Note: The server-side `StreamBufferCapacity` option (defaults to 10) can be configured to limit how many
+            // stream items from the client (per stream) will be stored before reading any more stream items (thus applying backpressure).
+            while (position < data.byteLength) {
+                const nextChunkSize = Math.min(chunkSize, data.byteLength - position);
```

Comment:
May want to consider a max chunk size if you care about interactivity. If the limit is long.MaxValue and you're sending a 1GB file it might take a while for the whole thing to be transferred and in the meantime nothing else can happen.

Reply:
The max chunk size already has an upper limit of the max SignalR message size.

Are you concerned about the case where someone has set the max SignalR message size to a large value or unlimited? If so I agree, we should check that we're not naively using just any value for this.

Reply:
> Are you concerned about the case where someone has set the max SignalR message size to a large value or unlimited?

Yes, that's why I said long.MaxValue

Reply:
Makes sense! I misread "limit" in your original comment to mean "the limit we're computing here" instead of "the configured SignalR message size limit".

Reply:
Thanks Brennan, I've updated to take the min of the maxIncomingBytes and 50 kb. Please do let me know if there are concerns with this cap (ie. perhaps we should go a 100kb?).